### ISSUE TITLE
Active item is lost when a pin or component is removed thru filter or item click.

### ISSUE THREAD

**Issue by juliocri** [Board-Explorer/board-explorer#16]
The active item is missed when multi-components are focused and an item from the list is removed.

Steps to reproduce:

1. Select multiple components (CTRL + click).
2. Deselect a single component with (CTRL + click).

Adding a new component add the just clicked element to the active var, but removing, removes all.

this issue is intended to add a new way to keep an active item after change the selected components.

**Comment by juliocri** [comment]
@jketreno  do you think this could help? (is still valid?) since the last feedback I don't know if the multiple addition of filters will be operating.

**Comment by jketreno** [comment]
The multi-select code was moved from being in board-explorer to in board-viewer itself to make it more of a "drop it in and it just works" control. When I tested the CTRL+click it was working to add/remove items from the URL and from the list of activated items in the board-viewer.

The two parts that weren't working:

1. When you stop pressing CTRL, the board-viewer should turn on the shadow (multiSelectActive should go false) -- to fix that I think we need to add a keyup listener on board-viewer and watch for CTRL to be released--if so, set mutliSelectActive to false and call _renderShadow.
2. I didn't get a chance to finish moving over the code you had added to support de-selecting all contained pins of a component if a component is removed (in testing that code in the old version, I notice that CTRL-click on VBAT to remove it it will throw an exception--VBAT is a component, but not all components have pins; the code doesn't check for item.pins before walking the array)
